### DISCUSSION TITLE
Help to set correct error pattern.

### DISCUSSION THREAD

**Discussion by guaracy** [orbitalquark/textadept#226] [Q&A] [answered]
The [Arturo](https://arturo-lang.io/) language returns errors in several lines. 
I tested in lua and appeared to work ok.

```
s = [[
>> Runtime | File: test.art
     error | Line: 3
           | 
           | symbol not found: lop
           | perhaps you meant... pop ?
           |                or... loop ?
           |                or... log ?
]]
re = '.*File: (.-)%s.*Line: (%d+).-| .-| (.-)\n'
p,l,e = string.match(s,re)
print('program: ',p)
print('   line: ',l)
print('  error: ',e)
```
result:
```
> cd /home/guaracy/projects/lua
> lua -e "io.stdout:setvbuf('no')" "teste.lua"
program: 	test.art
   line: 	3
  error: 	symbol not found: lop
> exit status: 0
```

The follow line in init.lua does not work.

textadept.run.error_patterns.arturo = {'.*File: (.-)%s.*Line: (%d+).-| .-| (.-)\n'} 

Am i missing something or this cannot be done.

TIA

**Comment by orbitalquark** [comment]
'.*' is greedy in Lua's pattern matching language. Try '.-' instead.

**Comment by guaracy** [comment]
I think textadept have a problem with multiline errors. Does not working with Julia too. Can you confirm?

Anyway i can catch only the filename and, when doubleclick, open the file and a window to go to line number.

**Comment by orbitalquark** [comment, marked as answer]
You are correct. Textadept cannot match multi-line error messages. Sorry I missed that in your original post :(
